File: objectWithTypeScript/src/main/Chapter01/Audience.ts
import { Bag } from './Bag';
import { Ticket } from './Ticket';

export class Audience {
  private _bag: Bag;

  constructor(bag: Bag) {
    this._bag = bag;
  }

  get bag(): Bag {
    return this._bag;
  }

  buy(ticket: Ticket): number {
    return this._bag.hold(ticket);
  }
}

File: objectWithJava/src/main/com/chapter01/Audience.java
package com.chapter01;

public class Audience {
  private Bag bag;
  
  public Audience(Bag bag) {
    this.bag = bag;
  }
  
  public Bag getBag() {
    return bag;
  }
  
  public Long buy(Ticket ticket) {
    return this.bag.hold(ticket);
  }
}
